### PR TITLE
Experimental Jest support

### PR DESCRIPTION
**Pull request by wooly**

### CODE REVIEW

File: internal/runner/jest.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	return files, nil
+}
+
+func (Jest) discoveryPattern() DiscoveryPattern {
+	includePattern := os.Getenv("BUILDKITE_SPLITTER_TEST_FILE_PATTERN")
+
+	if includePattern == "" {
+		includePattern = "**/*.spec.tsx"
```
Comment (nprizal) [inline internal/runner/jest.go:49]: this is the default file pattern `[ "**/__tests__/**/*.[jt]s?(x)", "**/?(*.)+(spec|test).[jt]s?(x)" ]`
https://jestjs.io/docs/configuration#testmatch-arraystring

Comment (nprizal) [reply internal/runner/jest.go:49]: Tried to convert that default into glob syntax that could be supported by zzglob
`**/{__tests__/**/*,*.spec,*.test}.{ts,js,jsx,tsx}`
[Glob test](https://globster.xyz/?q=**%2F%7B__tests__%2F**%2F*%2C*.spec%2C*.test%7D.%7Bts%2Cjs%2Cjsx%2Ctsx%7D&f=%2Fmyapp%2Fsomething%2F__tests__%2Fsome%2Ftest.jsx%2C%2Fmyapp%2Fsomething%2F__tests__%2Fsome%2Fother%2Ftest.js%2C%2Fmyapp%2Fsomething%2F__tests__%2Ftest.tsx%2C%2Fmyapp%2F__tests__%2Fsome%2Fsome%2Ftest.ts%2C%2Fmyapp%2Ftest.spec.ts%2C%2Fmyapp%2Fsomething%2Ftest.spec.tsx%2C%2Fmyapp%2Fsomething%2Fsome%2Ftest.spec.jsx%2C%2Fmyapp%2Fsomething%2Fsome%2Fother%2Ftest.spec.js%2C%2Fmyapp%2Fsomething%2Fsome%2Fnot_a_test.js%2C%2Fmyapp%2Fsomething%2Fsome%2Fmy.test.js)

File: internal/runner/jest.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	if len(files) == 0 {
+		return nil, fmt.Errorf("no files found with pattern %q and exclude pattern %q", j.TestFilePattern, j.TestFileExcludePattern)
+	}
+
+	return files, nil
+}
+
+func (j Jest) Command(testCases []string) (*exec.Cmd, error) {
```
Comment (nprizal) [inline internal/runner/jest.go:50]: I'm thinking of removing `Command` and `RetryCommand` and introduce `Run` to the runner interface that will execute the command, as well as parsing the result.
That will also enable us to unit test the command execution.
I have a draft [PR](https://github.com/buildkite/test-splitter/pull/156) for reference.

**Comment by wooly** [comment]
Superseded by #166

**Pull request closed without merging**